packages/babel-plugin-ember-test-metadata/utils.js

```diff
@@ -33,26 +34,76 @@ function getNodeProperty(node, path) {
 }
 
 /**
- * Get a normalized file path. If Embroider prefix is present, strip it out
+ * Function to parse defaults.project and only return info to be used by the plugin
+ * @param {object} project Ember defaults.project
+ * @returns {object} Contains project name and ember-addon path info
+ */
+function getProjectInfo(project) {
+  const parsedProjectInfo = {};
+
+  if (project) {
+    parsedProjectInfo.pkg = {
+      name: project.pkg.name,
+      'ember-addon': {
+        paths: project.pkg['ember-addon'].paths,
+      },
+    };
+  }
+
+  return parsedProjectInfo;
+}
+
+/**
+ * Strip out project name prefix segement(s) from file path
+ * @param {array} tokens File path segments
+ * @param {object} projectInfo Contains project name, ember-addon path info
+ * @returns {string} A joined file path, e.g. tests/acceptance/my-test.js
+ */
+function getParsedClassicFilepath(tokens, projectInfo) {
+  const projectNameTokens = projectInfo.pkg.name.split(path.sep);
+
+  tokens.splice(0, tokens.indexOf(projectNameTokens[0]) + projectNameTokens.length);
+
+  return tokens.join(path.sep);
+}
+
+/**
+ * Strip out Embroider prefix (embroider/nnnnnn) segements from file path
+ * @param {array} tokens File path segments
+ * @returns {string} A joined file path, e.g. tests/acceptance/my-test.js
+ */
+function getParsedEmbroiderFilepath(tokens) {
+  const RELATIVE_PATH_ROOT = 2;
+
+  tokens.splice(0, tokens.lastIndexOf(EMBROIDER) + RELATIVE_PATH_ROOT);
+
+  return tokens.join(path.sep);
+}
+
+/**
+ * Get a normalized file path, based on whether the app build is classic or with Embroider
  * @param {object} fileOpts Babel state.file.opts which include root and filename props
+ * @param {object} projectInfo Contains project name, ember-addon path info
  * @returns {string} E.g. tests/acceptance/my-test.js
  */
-function getNormalizedFilePath(fileOpts) {
+function getNormalizedFilePath(fileOpts, projectInfo) {
   let { root, filename } = fileOpts;
   const tokens = filename.split(path.sep);
-  const EMBROIDER = 'embroider';
+  const isEmbroider = tokens.includes(EMBROIDER);
 
-  if (tokens.includes(EMBROIDER)) {
-    const RELATIVE_PATH_ROOT = 2;
-
-    tokens.splice(0, tokens.lastIndexOf(EMBROIDER) + RELATIVE_PATH_ROOT);
+  if (isEmbroider) {
+    filename = getParsedEmbroiderFilepath(tokens);
+  } else {
+    filename = getParsedClassicFilepath(tokens, projectInfo);
   }
-  filename = tokens.join(path.sep);
 
   return path.relative(root, filename);
 }
 
 module.exports = {
   getNodeProperty,
   getNormalizedFilePath,
+  getProjectInfo,
+  getParsedClassicFilepath,
```

Comment:
I don't think these actually need to be exported, since they're only used in this file, right? they're effectively private methods.

Reply:
If they're tested (which they should be, they need to be exported).

Reply:
Yes, they are tested, so yes, needed to export.

Reply:
Would it be weird to prefix them with an underscore, since they are private, and export them as such, for testing?

Reply:
It wouldn't be weird at all, though I'm not sure that's necessary. They're in a utils module which someone would have to dig for if they wanted to use them. Also, does it really matter?

Reply:
I went ahead and gave them underscores

Reply:
IMO, tests should exercise the public API; corollary there is that if you are testing it, it is in the public API.

Reply:
So you're saying no private API should be tested? That means we should only be testing at the higher level, essentially acceptance testing?

FWIW, this pattern is used heavily throughout a lot of our node/Ember ecosystem. I can point you to references if you'd like.

Reply:
unless there's undue difficulty in exercising the internals of a public API, then yes, I avoid testing private methods. There are advantages to keeping private methods private, and as a result this is not a unique or terribly controversial opinion, but I'm not rigid about it either.

Reply:
Sync'd with Cam during our regular sync and we are ok to close this convo point. Appreciate all the input and insightful perspective.